src/server/game/Entities/Object/Object.cpp

```diff
@@ -509,111 +509,165 @@ void Object::BuildMovementUpdate(ByteBuffer* data, uint32 flags) const
+        bool hasTargetRollPitchYaw  = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_TARGET_ROLL_PITCH_YAW);
+        bool hasScaleCurveID        = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_SCALE_CURVE);
+        bool hasMorphCurveID        = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_MORPH_CURVE);
+        bool hasFacingCurveID       = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_FACING_CURVE);
+        bool hasMoveCurveID         = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_MOVE_CURVE);
+        bool hasUnk2                = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_UNK2);
+        bool hasUnk3                = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_UNK3);
+        bool hasUnk4                = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_UNK4);
+        bool hasAreaTriggerSphere   = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_SPHERE);
+        bool hasAreaTriggerBox      = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_BOX);
+        bool hasAreaTriggerPolygon  = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_POLYGON);
+        bool hasAreaTriggerCylinder = areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_HAS_CYLINDER);
+        bool hasAreaTriggerSpline   = false;// areaTrigger->HasFlag(AREATRIGGER_FLAG_HAS_SPLINE);
+        bool hasAreaTriggerUnkType  = false;// areaTrigger->HasFlag(AREATRIGGER_FLAG_UNK5);
+
+        data->WriteBit(hasAbsoluteOrientation);
+        data->WriteBit(hasDynamicShape);
+        data->WriteBit(hasAttached);
+        data->WriteBit(hasFaceMovementDir);
+        data->WriteBit(hasFollowsTerrain);
+        data->WriteBit(hasUnk1);
+        data->WriteBit(hasTargetRollPitchYaw);
+        data->WriteBit(hasScaleCurveID);
+        data->WriteBit(hasMorphCurveID);
+        data->WriteBit(hasFacingCurveID);
+        data->WriteBit(hasMoveCurveID);
+        data->WriteBit(hasUnk2);
+        data->WriteBit(hasUnk3);
+        data->WriteBit(hasUnk4);
+        data->WriteBit(hasAreaTriggerSphere);
+        data->WriteBit(hasAreaTriggerBox);
+        data->WriteBit(hasAreaTriggerPolygon);
+        data->WriteBit(hasAreaTriggerCylinder);
+        data->WriteBit(hasAreaTriggerSpline);
+        data->WriteBit(hasAreaTriggerUnkType);
+
+        if (areaTrigger->GetTemplate()->HasFlag(AREATRIGGER_FLAG_UNK3))
+            data->WriteBit(0);
+
+        if (hasAreaTriggerSpline)
+        {
+            //AreaTriggerHandler.ReadAreaTriggerSpline(packet, index);
+        }
 
-    //    if (HasTargetRollPitchYaw)
-    //        *data << Vector3(TargetRollPitchYaw);
+        if (hasTargetRollPitchYaw)
+        {
+            *data << float(0); // X
+            *data << float(0); // Y
+            *data << float(0); // Z
+        }
 
-    //    if (HasScaleCurveID)
-    //        *data << uint32(ScaleCurveID);
+        if (hasScaleCurveID)
+            *data << areaTriggerTemplate->ScaleCurveId;
```

Comment:
always explicitly specify the type when appending to packets (ObjectGuid is the only exception)
`*data << uint32(areaTriggerTemplate->ScaleCurveId);`